dashboard/src/components/Header/NamespaceSelector.test.tsx

```diff
@@ -63,12 +66,35 @@ it("renders the default namespace option if no namespaces provided", () => {
   const wrapper = shallow(<NamespaceSelector {...props} />);
   const select = wrapper.find(".NamespaceSelector__select").first();
 
-  expect(select.props()).toEqual(
-    expect.objectContaining({
-      options: [
-        { label: "All Namespaces", value: "_all" },
-        { label: defaultProps.defaultNamespace, value: defaultProps.defaultNamespace },
-      ],
-    }),
+  expect(select.props()).toMatchObject({
+    options: [
+      { label: "All Namespaces", value: "_all" },
+      { label: defaultProps.defaultNamespace, value: defaultProps.defaultNamespace },
+      { label: "Create New", value: "new" },
```

Comment:
I'm wondering if we need to show the option to "Create New" namespace if we know the user cannot do this. One simple, but not perfect, way is that we could only show the option to create a new namespace when there are more than 2 options in the selector already. It's not perfect as it only indicates the user can read namespaces, but is trivial to implement (EDIT: Actually, forget that, as we can assign RBAC to write namespaces without read, I assume). Not sure if it's worth considering returning this info when getting namespaces, but either way, it'd be another PR. See what you think.

Reply:
IMO hiding things may lead to a worse UX. If I am an unprivileged user and I want to create a namespace, I get a clear error that I don't have enough permissions to do so. If we hide the option it's not clear if I cannot do it or if it's the tool the one that doesn't support it.

Reply:
Yep, good point.